src/pubsub.js

```diff
@@ -271,22 +235,31 @@ class BasicPubSub extends Pubsub {
       throw new Error('Pubsub has not started')
     }
 
+    // normalize input and remove existing subscriptions
     topics = utils.ensureArray(topics)
-
     const newTopics = topics.filter((topic) => !this.subscriptions.has(topic))
     if (newTopics.length === 0) {
       return
     }
+    this._subscribe(newTopics)
```

Comment:
Why creating a new function? I think we will not reuse it anywhere else and I don't think it removes complexity.

Reply:
Its reused in src/index.js right now to hook into the "list of new topics" to subscribe to. The idea is that `subscribe` directly receives user input, munges into a canonical form.  `_subscribe` can then be overridden (always calling super._subscribe) and topics can be consumed there without this additional step of filtering/munging of inputs.
We could remove it, but then we will need to do the above filtering again if `subscribe` is overridden. Or perhaps we can have `subscribe` return the list of filtered topics, so overriding functions can still have access.

Reply:
I was thinking about just getting the `subscribe` overridden. I did not see that join needed the topics filtered. I am ok with going like you currently have. I think we should not change the API return values